### PR TITLE
test: e2e no floating promises [INFENG-668]

### PR DESCRIPTION
**Pull request by JComins000**
## Ticket

[INFENG-668]

## Description

show linting errors on floating promises

## Test Plan

Here's [an example](https://app.circleci.com/pipelines/github/determined-ai/determined/55037/workflows/3a8e35d1-7d2e-46a0-b9c7-9e49e6c92e41/jobs/2525620?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary&invite=true#step-113-297_153) of a successful lint

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[INFENG-668]: https://hpe-aiatscale.atlassian.net/browse/INFENG-668?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by djanicekpach — APPROVED** [review]
I like this as long as it doesn't cause problems for non-test frontend code.

File: webui/react/src/e2e/models/hew/DataGrid.ts [hunk 1]
```diff
@@ -290,7 +290,7 @@ export class HeadRow extends NamedComponent {
           if (index === null) throw new Error();
           this.#columnDefs.set(await cell.innerText(), +index);
         } catch {
-          Promise.reject(
+          await Promise.reject(
```
Comment (ashtonG) [inline webui/react/src/e2e/models/hew/DataGrid.ts:293]: why not just throw the error?

Comment (JComins000) [reply webui/react/src/e2e/models/hew/DataGrid.ts:293]: oh, sure!

Comment (JComins000) [reply webui/react/src/e2e/models/hew/DataGrid.ts]: check again, please!

**Review by ashtonG — APPROVED** [review]

**Pull request merged**